py/configu/commands/upsert_command.py

```diff
@@ -66,37 +66,49 @@ def run(self):
                 raise ValueError(
                     error_message(
                         f"invalid config key '{key}'",
-                        scope_location,
+                        error_scope,
                         f"key '{key}' must be declared on schema {schema.path}",
                     )
                 )
             if value and cfgu.template is not None:
                 raise ValueError(
                     error_message(
                         f"invalid assignment to config key '{key}'",
-                        scope_location,
+                        error_scope,
                         "keys declared with template mustn't have a value",
                     )
                 )
-            type_test = ConfigSchema.CFGU.VALIDATORS.get(cfgu.type.value, lambda: False)
-            test_values = (
-                (
-                    value,
-                    cfgu.pattern,
+            try:
+                type_test = ConfigSchema.CFGU.VALIDATORS[cfgu.type.value]
+                test_values = (
+                    (
+                        value,
+                        cfgu.pattern,
+                    )
+                    if cfgu.type == CfguType.REG_EX
+                    else (value,)
                 )
-                if cfgu.type == CfguType.REG_EX
-                else (value,)
-            )
-            if value and not type_test(*test_values):
-                of_type = cfgu.type.value
-                if cfgu.type == CfguType.REG_EX:
-                    of_type += f"({cfgu.pattern})"
-                raise ValueError(
-                    error_message(
-                        f"invalid config value '{value}' for key '{key}'",
-                        scope_location,
-                        f"value '{value}' must be of type '{of_type}'",
+                if value and not type_test(*test_values):
+                    of_type = cfgu.type.value
+                    if cfgu.type == CfguType.REG_EX:
+                        of_type += f"({cfgu.pattern})"
+                    raise ValueError(
+                        error_message(
+                            f"invalid config value '{value}' for key '{key}'",
+                            error_scope,
+                            f"value '{value}' must be of type '{of_type}'",
+                        )
                     )
+            except KeyError:
+                raise KeyError(
```

Comment:
`raise KeyError(...) from e` pattern. Also removal of extra f-strings if possible.

Reply:
done